### PR TITLE
Adding basic elemental analysis 2 GUI

### PR DESCRIPTION
**Pull request by AnthonyLim23**
This PR adds the basic code for a new elemental analysis GUI. The GUI itself just has some dummy line edits for now.

**To test:**
1. Before starting Mantid go to https://github.com/mantidproject/mantid/blob/master/Framework/Properties/Mantid.properties.template#L24 and add `Muon/Elemental_Analysis_2.py` to the end of the line

2. Start Mantid

3. Open the elemental analysis 2 GUI - it should appear. 

4. Check the manage user directories button works. Everything else is "dead"

5. Close the GUI and check it doesn't crash

There is no associated issue.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by joseph-torsney** [comment]
Is the gui supposed to appear in the interfaces menu after doing step 1? - It's not appearing for me

**Review by joseph-torsney — DISMISSED** [review]
Looks good :+1:

**Review by martyngigg — CHANGES_REQUESTED** [review]
As this is starting fresh I'd like to consider that we write some integration tests using [QTest](https://doc.qt.io/qt-5/qtest.html). For the basic case here we can just test that the MainWindow can be created successfully?

I'd also like to suggest that we start trying to maintain an acceptance test document in https://developer.mantidproject.org/Testing/index.html where we can add cases that cannot be automatically captured. Hopefully this two pronged approach will improve our ability to do real-world testing on the GUIs.

**Comment by AnthonyLim23** [comment]
@martyngigg that sounds good, but I do not know anything about how to use QTest. Is there some good documentation on using it?

**Comment by martyngigg** [comment]
@AnthonyLim23 :+1: The Python side functions are a pretty good one-to-one map from C++ so the  [C++ tutorial](https://doc.qt.io/qt-5/qttestlib-tutorial3-example.html) is a good start to get the idea. [This tutorial](http://johnnado.com/pyqt-qtest-example/) also shows a nice example of using it in PyQt - we'll just need to switch out the PyQt4 for qtpy.

File: dev-docs/source/Testing/ElementalAnalysis/ElementalAnalysisTests.rst [hunk 1]
```diff
@@ -0,0 +1,26 @@
+.. _elemental_analysis_gui_testing:
```
Comment (martyngigg) [inline dev-docs/source/Testing/ElementalAnalysis/ElementalAnalysisTests.rst:1]: Nice! A good thing we can build on :+1: 

I wonder if we should update the PR template as a reminder to reviewers to check whether these types of document should be updated.

**Review by martyngigg — APPROVED** [review]

**Comment by martyngigg** [comment]
Unrelated test failure.

File: dev-docs/source/Testing/ElementalAnalysis/ElementalAnalysisTests.rst (hunk 1 above)
Comment (AnthonyLim23) [reply dev-docs/source/Testing/ElementalAnalysis/ElementalAnalysisTests.rst:1]: probably worth doing along with a reminder for QTest (if possible)

**Pull request merged**